### PR TITLE
Fix bug where parameters could not be toggled off

### PR DESCRIPTION
**Pull request by dannysongg**
## Description

Some parameters for the Publish-AzWebApp cmdlet need to be able to be set to false. The current parameters are of type SwitchParameters so they can only be set to true. This change will change some parameters to the type nullable boolean. This is a bug fix.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [X] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Websites/Websites/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Allow certain parameters to be toggled off for `Publish-AzWebApp``
```
Comment (YanaXu) [inline src/Websites/Websites/ChangeLog.md:21]:
```suggestion
* Allowed the certain parameters to be toggled off for `Publish-AzWebApp``
```

**Comment by YanaXu** [comment]
Hi @dannysongg, this PR introduce breaking changes which were not pre-announced in the former 2 releases. Please take the 2 PRs for example: [Add upcoming Breaking Change message](https://github.com/Azure/azure-powershell/pull/24424) and  [the breaking change update](https://github.com/Azure/azure-powershell/pull/24675).

First, you have add the upcoming breaking change message in [Az 12.1.0 (2024-07-02)](https://github.com/Azure/azure-powershell/milestone/160). 
Second, you can submit this PR in Az 13.0.0.

**Comment by dannysongg** [comment]
@YanaXu this is a bug fix that fixes a CRI. Cx needs change released ASAP

**Comment by YanaXu** [comment]
Hi @dannysongg, please confirm this: it's a bug fix, which means the cmdlet is not working before. Is this correct?
If yes, please suppress the breaking chagnes.
If no, it's a breaking change and we need to hold it.

**Comment by YanaXu** [comment]
> Hi @dannysongg, please confirm this: it's a bug fix, which means the cmdlet is not working before. Is this correct? If yes, please suppress the breaking chagnes. If no, it's a breaking change and we need to hold it.

Hi @dannysongg, please confirm this comment. FYI, the code freeze date is next Monday.

**Comment by dannysongg** [comment]
@YanaXu yes this is a bug fix. This is fixing an unintended behavior reported by Cx. I will suppress the breaking change.

**Pull request closed without merging**